travis build fails intermittently with postgres error

this seems to happen randomly:

``` bash
before_script.1
0.33s$ psql -c 'create database forum;' -U postgres
CREATE DATABASE
before_script.2
0.22s$ psql -c 'create database forum_test;' -U postgres
CREATE DATABASE
2.40s$ npm test
> forum.node@0.0.0 test /home/travis/build/twinlabs/forum
> NODE_ENV=test mocha --require test/test_helper.js --recursive ./test
   info  - socket.io started
  listening on port 3000
․․․․․․․․․․․․․․․․․․Executing (default): DROP TABLE IF EXISTS "post" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "post" ("id"  SERIAL , "body" TEXT, "user_id" INTEGER NOT NULL, "parent" INTEGER, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL, "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
Executing (default): DROP TABLE IF EXISTS "forum_user" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "forum_user" ("id"  SERIAL , "name" VARCHAR(255) NOT NULL, "email" VARCHAR(255), "password" VARCHAR(255), "created_at" TIMESTAMP WITH TIME ZONE NOT NULL, "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
Executing (default): INSERT INTO "post" ("body","user_id","created_at","updated_at") VALUES ('it''s happening',1,'2014-09-01 12:26:15.693 +00:00','2014-09-01 12:26:15.693 +00:00') RETURNING *;
․Executing (default): INSERT INTO "forum_user" ("id","name","created_at","updated_at") VALUES (7,'Ty','2014-09-01 12:26:15.701 +00:00','2014-09-01 12:26:15.701 +00:00') RETURNING *;
Executing (default): INSERT INTO "post" ("id","body","user_id","created_at","updated_at") VALUES (77,'hello world. i am ty.',7,'2014-09-01 12:26:15.701 +00:00','2014-09-01 12:26:15.701 +00:00') RETURNING *;
Executing (default): INSERT INTO "post" ("id","body","user_id","created_at","updated_at") VALUES (78,'anyone want to post with me?',7,'2014-09-01 12:26:15.702 +00:00','2014-09-01 12:26:15.702 +00:00') RETURNING *;
Executing (default): SELECT * FROM "post" WHERE "post"."user_id"=7;
․․․․․․Executing (default): DROP TABLE IF EXISTS "post" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "post" ("id"  SERIAL , "body" TEXT, "user_id" INTEGER NOT NULL, "parent" INTEGER, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL, "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
Executing (default): DROP TABLE IF EXISTS "forum_user" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "forum_user" ("id"  SERIAL , "name" VARCHAR(255) NOT NULL, "email" VARCHAR(255), "password" VARCHAR(255), "created_at" TIMESTAMP WITH TIME ZONE NOT NULL, "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
․․Executing (default): INSERT INTO "post" ("user_id","created_at","updated_at") VALUES (1,'2014-09-01 12:26:15.860 +00:00','2014-09-01 12:26:15.860 +00:00') RETURNING *;
․
  25 passing (266ms)
  2 pending
  1 failing
  1) authentication "before all" hook:
     Uncaught error: relation "post" does not exist
      at Connection.parseE (/home/travis/build/twinlabs/forum/node_modules/pg/lib/connection.js:561:11)
      at Connection.parseMessage (/home/travis/build/twinlabs/forum/node_modules/pg/lib/connection.js:390:17)
      at Connection.attachListeners (/home/travis/build/twinlabs/forum/node_modules/pg/lib/connection.js:92:20)
      at Socket.EventEmitter.emit (events.js:96:17)
      at TCP.onread (net.js:397:14)
npm ERR! weird error 1
npm ERR! not ok code 0
The command "npm test" exited with 1.
Done. Your build exited with 1.
```

--- a/config/environments.js
+++ b/config/environments.js
@@ -10,7 +10,6 @@ module.exports = function(app){
   });
 
   app.configure('test', function(){
-    process.env.DATABASE_URL = "postgres://postgres@localhost/forum_test";
     // test environment-specific stuff here:
   });
 

--- a/test/test_helper.js
+++ b/test/test_helper.js
@@ -1,2 +1,6 @@
+if (process.env.NODE_ENV === 'test') {
+  process.env.DATABASE_URL = "postgres://postgres@localhost/forum_test";
+}
+
 require('../lib/helpers');
 

